src/dotnet-install.sh

```diff
@@ -341,6 +344,30 @@ get_normalized_architecture_from_architecture() {
     return 1
 }
 
+# args:
+# user_defined_os - $1
+get_normalized_os() {
+    eval $invocation
+
+    local osname="$(to_lowercase "$1")"
+    if [ ! -z "$osname" ]; then
+        case "$osname" in
+            osx | freebsd | rhel.6 | linux-musl | linux)
+                echo "$osname"
+                return 0
+                ;;
+            *)
+                say_warning "'$user_defined_os' is not a valid platform. The platform will be evaluated based on machine configuration."
```

Comment:
I chose to fallback to auto-detection in case choice provided by user is not supported, however open for suggestions here.